templates/test/ci/prow-topology/base.yaml

```diff
@@ -0,0 +1,264 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: ClusterClass
+metadata:
+  name: ci-default
+spec:
+  controlPlane:
+    ref:
+      apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+      kind: KubeadmControlPlaneTemplate
+      name: ci-default-kubeadm-control-plane
+    machineInfrastructure:
+      ref:
+        kind: AzureMachineTemplate
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        name: ci-default-control-plane
+  infrastructure:
+    ref:
+      apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+      kind: AzureClusterTemplate
+      name: ci-default-azure-cluster
+  workers:
+    machineDeployments:
+      - class: ci-default-worker
+        template:
+          bootstrap:
+            ref:
+              apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+              kind: KubeadmConfigTemplate
+              name: ci-default-worker
+          infrastructure:
+            ref:
+              apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+              kind: AzureMachineTemplate
+              name: ci-default-worker
+        machineHealthCheck:
+          maxUnhealthy: 100%
+          unhealthyConditions:
+            - type: E2ENodeUnhealthy
```

Comment:
Where does this value come from? I am looking into adding a health check for windows on the other e2e test but can not determine if this health condition is valid.  I don't find this type in capz/capi repositories and the docs give examples using `Ready` on the machines: https://cluster-api.sigs.k8s.io/tasks/healthcheck.html#creating-a-machinehealthcheck

Using `Ready` on the machines makes sense but I wasn't sure if I was missing something

Reply:
I took it from the [existing templates](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/f0c71f92a1d35805eafc03a31caaf1d5933f50ea/templates/test/ci/cluster-template-prow.yaml#L330), but not sure how it works though.